src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -17,18 +38,34 @@ export class TransactionApi {
     // TODO key is not final
     const cacheKey = `balances-${this.chainId}-${safeAddress}-${trusted}-${excludeSpam}`;
     const url = `${this.baseUrl}/api/v1/safes/${safeAddress}/balances/usd/`;
-    return await this.dataSource.get<Balance[]>(cacheKey, url, {
+    const balances: [] = await this.dataSource.get(cacheKey, url, {
       params: {
         trusted: trusted,
         excludeSpam: excludeSpam,
       },
     });
+
+    if (!balances.every((balance) => this.isValidBalance(balance))) {
+      // TODO: probably we want to invalidate cache at this point
+      const errors = this.isValidBalance.errors as DefinedError[];
+      throw this.validationErrorFactory.from(errors);
+    }
```

Comment:
We should test when `balance` is not valid.

Reply:
Added controller tests in [1a50781](https://github.com/5afe/safe-client-gateway-nest/pull/66/commits/1a5078151be359884f206f1691f9feef39010174)